bittide/src/Clash/Cores/Xilinx/Gth/Internal.hs

```diff
@@ -0,0 +1,343 @@
+  , rxctrl3Out :: Signal rxUser2 (BitVector 8)
+  -- ^ @rxctrl3_out@
+  }
+
+type GthCore txUser txUser2 rxUser rxUser2 refclk0 freerun txS rxS =
+  ( KnownDomain txUser
+  , KnownDomain txUser2
+  , KnownDomain rxUser
+  , KnownDomain rxUser2
+  , KnownDomain refclk0
+  , KnownDomain freerun
+  , KnownDomain txS
+  , KnownDomain rxS
+  ) =>
+  CoreInput txUser txUser2 rxUser rxUser2 refclk0 freerun rxS ->
+  CoreOutput txUser txUser2 rxUser rxUser2 txS
+
+gthCore ::
+  forall txUser txUser2 rxUser rxUser2 refclk0 freerun txS rxS.
+  GthCore txUser txUser2 rxUser rxUser2 refclk0 freerun txS rxS
+gthCore input
+  | clashSimulation = simCore
+  | otherwise = synthCore
+ where
+  simCore =
+    CoreOutput
+      { gthtxOut = SimOnly input.gtwizUserdataTxIn
+      , gthtxnOut = pure 0
+      , gthtxpOut = pure 0
+      , txoutclkOut = clockGen
+      , rxoutclkOut = clockGen
+      , gtwizUserdataRxOut = unSimOnly input.gthrxIn
+      , gtwizResetTxDoneOut = pure 1
+      , gtwizResetRxDoneOut = pure 1
+      , txpmaresetdoneOut = pure 1
+      , rxpmaresetdoneOut = pure 1
+      , rxctrl0Out = pure 0
+      , rxctrl1Out = pure 0
+      , rxctrl2Out = pure 0
+      , rxctrl3Out = pure 0
+      }
+
+  synthCore =
+    CoreOutput
+      { gthtxOut = SimOnly input.gtwizUserdataTxIn
+      , gthtxnOut = gthtxn_out
+      , gthtxpOut = gthtxp_out
+      , txoutclkOut = txoutclk_out
+      , rxoutclkOut = rxoutclk_out
+      , gtwizUserdataRxOut = gtwiz_userdata_rx_out
+      , gtwizResetTxDoneOut = pack <$> gtwiz_reset_tx_done_out
+      , gtwizResetRxDoneOut = pack <$> gtwiz_reset_rx_done_out
+      , txpmaresetdoneOut = pack <$> txpmaresetdone_out
+      , rxpmaresetdoneOut = pack <$> rxpmaresetdone_out
+      , rxctrl0Out = rxctrl0_out
+      , rxctrl1Out = rxctrl1_out
+      , rxctrl2Out = rxctrl2_out
+      , rxctrl3Out = rxctrl3_out
+      }
+   where
+    ( unPort -> gthtxn_out
+      , unPort -> gthtxp_out
+      , unPort -> txoutclk_out
+      , unPort -> rxoutclk_out
+      , unPort -> gtwiz_userdata_rx_out
+      , unPort -> gtwiz_reset_tx_done_out
+      , unPort -> gtwiz_reset_rx_done_out
+      , unPort -> txpmaresetdone_out
+      , unPort -> rxpmaresetdone_out
+      , unPort -> rxctrl0_out
+      , unPort -> rxctrl1_out
+      , unPort -> rxctrl2_out
+      , unPort -> rxctrl3_out
+      ) = go
+
+    go ::
+      ( Port "gthtxn_out" txS (BitVector 1)
+      , Port "gthtxp_out" txS (BitVector 1)
+      , ClockPort "txoutclk_out" txUser
+      , ClockPort "rxoutclk_out" rxUser
+      , Port "gtwiz_userdata_rx_out" rxUser2 (BitVector 64)
+      , Port "gtwiz_reset_tx_done_out" txUser2 Bit
+      , Port "gtwiz_reset_rx_done_out" rxUser2 Bit
+      , Port "txpmaresetdone_out" txUser Bit
+      , Port "rxpmaresetdone_out" rxUser Bit
+      , Port "rxctrl0_out" rxUser2 (BitVector 16)
+      , Port "rxctrl1_out" rxUser2 (BitVector 16)
+      , Port "rxctrl2_out" rxUser2 (BitVector 8)
+      , Port "rxctrl3_out" rxUser2 (BitVector 8)
+      )
```

Comment:
This should be `@`d where `unPort` is called, otherwise we're suffering from positional hell again.